packages/github-actions/teams-toolkit-cicd/src/main.ts

```diff
@@ -1,19 +1,76 @@
 import * as core from '@actions/core'
-import {wait} from './wait'
+import {ActionInputs} from './constant'
+import {Operations} from './operations'
+import {OperationType} from './enums/operationTypes'
+import {WordsToList} from './utils/words-to-list'
+import {InputsError} from './errors'
+import {Capability} from './enums/capabilities'
+import {BaseError} from './base-error'
 
 async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
-    core.debug(`Waiting ${ms} milliseconds ...`) // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
+    let projectRoot = core.getInput(ActionInputs.ProjectRoot)
+    if (!projectRoot && process.env.GITHUB_WORKSPACE) {
```

Comment:
Can we precheck the projectRoot in case it is an invalid path.

Reply:
Didn't quite get it. What do you mean by `precheck`?  The checking logic is around line 18-22.

Reply:
I mean check whether the path is a valid folder. I saw that in ops, you directly check the folder joint with this value. If the path is invalid, then following existence check always return false.

Reply:
I get it. Thanks, I will check if the projectRoot is a valid path.